File: models/flightInfo.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

// FlightInfo stores information about speakers' flights (see models.Speaker).
type FlightInfo struct {
	ID       primitive.ObjectID `json:"id" bson:"_id"`
	Inbound  time.Time          `json:"inbound" bson:"inbound"`
	Outbound time.Time          `json:"outbound" bson:"outbound"`

	// Airport
	From string `json:"from" bson:"from"`

	// Airport
	To string `json:"to" bson:"to"`

	// Link to the flight information
	Link string `json:"link" bson:"link"`

	Bought bool `json:"bought" bson:"bought"`

	// Cost of the flight in cents (€)
	Cost int `json:"cost" bson:"cost"`

	Notes string `json:"notes" bson:"notes"`
}

File: frontend/lib/models/flightInfo.dart
import 'dart:convert';

class FlightInfo {
  final String id;
  final int event;
  final String speaker;
  final DateTime inbound;
  final DateTime outbound;
  final String from;
  final String to;
  final String link;
  final bool bought;
  final int cost;
  final String notes;

  FlightInfo(
      {required this.id,
      required this.event,
      required this.speaker,
      required this.inbound,
      required this.outbound,
      required this.from,
      required this.to,
      required this.link,
      required this.bought,
      required this.cost,
      required this.notes});

  factory FlightInfo.fromJson(Map<String, dynamic> json) {
    return FlightInfo(
      id: json['id'],
      event: json['event'],
      speaker: json['speaker'],
      inbound: DateTime.parse(json['inbound']),
      outbound: DateTime.parse(json['outbound']),
      from: json['from'],
      to: json['to'],
      link: json['link'],
      bought: json['bought'],
      cost: json['cost'],
      notes: json['notes'],
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'event': event,
        'speaker': speaker,
        'inbound': inbound,
        'outbound': outbound,
        'from': from,
        'to': to,
        'link': link,
        'bought': bought,
        'cost': cost,
        'notes': notes
      };

  @override
  String toString() {
    return json.encode(this.toJson());
  }
}
